### ISSUE TITLE
Upgrade `react-router-dom` dependency in the dashboard

### ISSUE THREAD

**Issue by alpeb** [linkerd/linkerd2#7252]
We're on `5.3.0` and would like to upgrade to `v6`, but that's a breaking change that requires manual intervention in our routes handling.

The React-router project has promised a compatibility layer to come up at some point that would avoid having to change things too much, but I feel that's a second-best solution and would prefer to have things written as per the best practice.

For anyone wanting to tackling this, we're glad to provide guidance on the overall setup of the dashboard web app in linkerd (also note that BUILD.md provides lots of details).

**Comment by hackpk** [comment]
I'd like to work on this. Also I need guidance as this will be my first issue.

**Comment by alpeb** [comment]
@hackpk Excellent. My recommendation is to first try installing linkerd in your local cluster, then try making it work with a separate web development server as explained on BUILD.md, so that you can see in realtime any changes you make to the web code. Feel free to ask any questions in the #contributors channel in our Slack.

**Comment by will-codes-things** [comment]
Is this still needed?

**Comment by kleimkuhler** [comment]
Yes @will-codes-things we're still interested in upgrading this. If you have any questions about contributing you can still go to the #contributors channel on the Linkerd Slack.

**Comment by acald-creator** [comment]
Hello there!

I wanted to reach out and see if anyone has started working on this?

If not, I would like to work on this issue.

**Comment by acald-creator** [comment]
In fact, I'm already working through the migration, and also I understand that this will require changes to the React class components, since the new version of `react-router-dom` is using hooks. I will raise some questions in the Slack channel if I have any.

**Comment by acald-creator** [comment]
From what I could tell this issue is still open but I'm currently working on this issue.

I didn't receive much feedback from Linkerd team.

**Comment by adleong** [comment]
Hi @acald-creator, is there any feedback in particular you're looking for or anything we can do to support you?

**Comment by acald-creator** [comment]
@adleong Thanks for responding! No not at this time. I'm hoping to have a PR next week. But I will reach out if I have a question.

**Comment by acald-creator** [comment]
bump

Wanted to give an update that I am still working on this. My plan was to have the PR last week, but got sidetracked.

**Comment by acald-creator** [comment]
I have a branch that I am working on, but there were some questions asked in the Slack channel which I would like to elaborate and answer in this issue or the PR I will provide.

https://github.com/acald-creator/linkerd2/tree/upgrade/react-router-dom

I want to gather all the info first and some steps taken to make this transition possible.

**Comment by acald-creator** [comment]
Still working this, but was delayed from being sick last week.

**Comment by Mish2j** [comment]
Hi there, I would like to work on this. Should I wait to be assigned or I can start working?

**Comment by acald-creator** [comment]
I was working on this, but got delayed with other things I needed to work on.

I would like to still work on this, but I think it would be great to have additional help.

The road to migrate from v5 to v6 can be incremental, but there are slight breaking changes in certain instances, especially the main Router portion. v6 have changed how the router is designed so that would require the most work to be done.

**Comment by adleong** [comment]
Hi @acald-creator 

Do you think it will be possible to upgrade to v6 without switching the project to React Hooks?  Based on my understanding, this would be a significant effort to migrate the whole project over so I think we'd prefer to avoid that unless it prevents us from upgrading.

**Comment by acald-creator** [comment]
@adleong 

Here is the reference to a closed issue regarding the support for Class Components. 

https://github.com/remix-run/react-router/issues/8146

Just skimming through, we could do a HOC component, and that will just wrap any of the Class components that utilize the `react-router-dom` library. 

And here is the official guide with a locked closed issue, https://github.com/remix-run/react-router/discussions/8753 that walks you through on how to migrate.

I can certainly help with migrating overall, but I can be involved later, if you prefer to avoid this.

**Comment by pavanjoshi914** [comment]
@adleong is this issue still open to be assigned?
